mrjob/py2.py

```diff
@@ -0,0 +1,170 @@
+"""Minimal utilities to make Python work for 2.6+ and 3.3+
+
+Strategies for making `mrjob` work across Python versions:
+
+Bytes vs. Unicode
+-----------------
+
+It's tempting to use `from __future__ import unicode_literals` and require
+that all non-byte strings be unicode. But that doesn't really make sense for
+Python 2, where str (bytes) and unicode can be used interchangeably.
+
+So really our string datatypes fall into two categories, bytes, and
+"strings", which are either ``str`` (i.e., bytes) or ``unicode`` in Python 2,
+and ``str`` (i.e. unicode) in Python 3.
+
+These things should always be bytes:
+
+- input data files
+  - use ``'b'`` when opening files: ``open(..., 'rb')``
+  - read data from ``sys.stdin.buffer`` in Python 3, not ``sys.stdin``
+- data encoded by protocols
+- data from subprocesses (this already happens by default)
+- log files parsed by mrjob
```

Comment:
Latin-1 is bullet-proof in the sense that it will never except, but it will likely be wrong if a non-ascii character appears. internet-encoding preserves the no-except behavior but tries utf-8 first.

This stuff doesn't matter much, just fyi.
Do what you want to maintain :)